Tests/UnitTests/Diagnostics/DiagnosticsFileHandlerTests.swift

```diff
@@ -0,0 +1,134 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  DiagnosticsFileHandlerTests.swift
+//
+//  Created by Cesar de la Vega on 2/4/24.
+
+import Foundation
+import Nimble
+@testable import RevenueCat
+import XCTest
+
+@available(iOS 13.0, macOS 10.15, tvOS 13.0, watchOS 6.2, *)
+class BaseDiagnosticsFileHandlerTests: TestCase {
+
+    fileprivate var fileHandler: FileHandler!
+    fileprivate var handler: DiagnosticsFileHandler!
+
+    override func setUp() async throws {
+        try await super.setUp()
+
+        try AvailabilityChecks.iOS13APIAvailableOrSkipTest()
+
+        self.fileHandler = try Self.createWithTemporaryFile()
+        self.handler = .init(self.fileHandler)
+    }
+
+    override func tearDown() async throws {
+        self.handler = nil
+
+        try await super.tearDown()
+    }
+
+}
+
+@available(iOS 13.0, macOS 10.15, tvOS 13.0, watchOS 6.2, *)
+class DiagnosticsFileHandlerTests: BaseDiagnosticsFileHandlerTests {
+
+    // MARK: - append
+
+//    func testAppendEventWithProperties() async throws {
+//        let content = DiagnosticsEvent(name: "HTTP_REQUEST_PERFORMED",
+//                                       properties: ["key": AnyEncodable("value")],
+//                                       timestamp: Date())
+//
+//        await self.handler.appendEvent(diagnosticsEvent: content)
+//
+//        let entries = await self.handler.getEntries()
+//        expect(entries[0]).to(equal(content))
+//    }
```

Comment:
Been spending a good amount of time trying to fix this and I don't know what's up but it's erroring comparing the dates:

```
failed - expected to equal <DiagnosticsEvent(diagnosticType: "event", name: "HTTP_REQUEST_PERFORMED", properties: ["key": RevenueCat.AnyEncodable(value: "value")], timestamp: 2024-04-03 10:53:19 +0000)>, got <DiagnosticsEvent(diagnosticType: "event", name: "HTTP_REQUEST_PERFORMED", properties: ["key": RevenueCat.AnyEncodable(value: "value")], timestamp: 2024-04-03 10:53:19 +0000)>
```

I removed the timestamp from the comparison and it started to work. I think it's related to:

```
The issue with the timestamp comparison you're encountering in your test case is a common problem when dealing with dates in unit tests, especially when serialization and deserialization are involved. The problem arises because when you serialize and then deserialize a Date, the resulting Date might not be equal to the original due to precision loss or differences in the date encoding and decoding strategies.
```

Reply:
Hmm can we encode the timestamp as milliseconds since epoch? (That's how we have it in android right now). Not sure if that would help here.

Reply:
I was doing that and it was encoding it with decimals `1712142586558.343` that's why I changed to iso. I think the decimals makes it clear that it's some precision issue

Reply:
I'm not sure that's it - I think there might be some data precision getting lost in the encoding / decoding. 

For example, if you do `expect(try content.jsonEncodedData).to(equal(try content.jsonEncodedData))`, that actually works. So something between appending and getting the entries is losing precision

Reply:
yeah, that'll be it - we're comparing the timestamp of the DiagnosticsEvent post-encoding (`entries[0].timestamp`) vs pre-encoding `content.timestamp`, and there'll be precision loss because the encoder doesn't take nanoseconds

Reply:
easiest fix: update this line to
```swift
        let encodedContent = try content.encodeAndDecode()
        expect(entries[0]).to(equal(encodedContent))
```

or even
```swift
        let encodedContent = try content.encodeAndDecode()
        expect(entries[0]) == encodedContent
```

Reply:
That indeed fixed it!!!! Thank you so much